### PR TITLE
Created custom input component to be used in Send forms for the amount.

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

Introduce a central `SendInputAmount` component for enhanced reuse and feature integration, such as the "Max" button functionality.

# Changes

- Created the `SendInputAmount` component.
- Refactored `SendAmount` and `IcSendAmount` to utilize the new component.

# Tests

Verified that the existing behaviors are maintained under local development conditions.

### CODE REVIEW

File: src/frontend/src/icp/components/send/IcSendAmount.svelte [hunk 1]
```diff
@@ -50,62 +43,42 @@
...
-
-		amountError = undefined;
+		return amountError;
 	};
 
-	const debounceValidate = debounce(validate);
+	const customValidations = [validate];
 
-	$: amount, fee, $ckBtcMinterInfoStore, $ckEthMinterInfoStore, debounceValidate();
+	let reactivityVariables: (unknown | undefined)[];
```
Comment (peterpeterparker) [inline src/frontend/src/icp/components/send/IcSendAmount.svelte:75]: have you thought about exposing the function `debounceValidate` to the parent? don't like the pattern?

```
$: fee, $ckBtcMinterInfoStore, $ckEthMinterInfoStore, input.debounceValidate()
```

if you rather like to use the array, can we maybe find a less generic name for the property?

File: src/frontend/src/lib/components/send/SendInputAmount.svelte [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+			if (validationError) {
+				return validationError;
+			}
+		}
+		return validationError;
+	};
+
+	const validate = () => {
+		if (invalidAmount(amount)) {
+			return;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/send/SendInputAmount.svelte:33]: the error shouldn't be resetted here?

File: src/frontend/src/lib/components/send/SendInputAmount.svelte [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+		const parsedValue = parseToken({
+			value: `${amount}`,
+			unitName: tokenDecimals
+		});
+
+		error = runValidations(parsedValue);
+	};
+
+	const debounceValidate = debounce(validate, debounceTime);
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/send/SendInputAmount.svelte:44]: this should not be dynamic in case the validation changes (`let` vs `const`)?

File: src/frontend/src/lib/components/send/SendInputAmount.svelte [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	export let tokenDecimals: number | undefined = undefined;
+	export let label: string = $i18n.core.text.amount;
+	export let placeholder: string = $i18n.core.text.amount;
+	export let customValidations: ((value: BigNumber) => Error | undefined)[] = [];
+	export let reactivityVariables: (unknown | undefined)[] = [];
+	export let debounceTime: number | undefined = 300;
+
+	let error: Error | undefined;
+
+	const runValidations = (value: BigNumber) => {
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/send/SendInputAmount.svelte:20]: note: I did not reviewed this function given that I suggest to pass only a single validator

File: src/frontend/src/lib/utils/token.utils.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { E8S_PER_ICP } from '$lib/constants/icp.constants';
```
Comment (peterpeterparker) [inline src/frontend/src/lib/utils/token.utils.ts:1]: This is unrelated to this new `SendInputAmount` cmp right? Can you provide it in a separate PR.

**Review by peterpeterparker — COMMENTED** [review]
Thanks! A few comments and questions.

File: dfx.json [hunk 6]
```diff
@@ -81,7 +81,7 @@
 		"ckbtc_kyt": {
 			"type": "custom",
 			"candid": "target/ic/ckbtc_kyt.did",
-			"wasm": "target/ic/ckbtc_kyt.wasm",
+			"wasm": "target/iyc/ckbtc_kyt.wasm",
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply dfx.json:84]: Reverted

File: scripts/send.tokens.sh [hunk 7]
```diff
@@ -3,7 +3,7 @@
 # Utility Script: Send Tokens to a Local User
 # Particularly handy after installing the canisters for the first time or being forced by dfx to clean the local state.
 
-PRINCIPAL=x4w27-so7wg-cudsa-yy7fh-wcpy5-njul4-q54tv-euzzi-tdnzz-ill46-zqe
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply scripts/send.tokens.sh:6]: Reverted

File: src/frontend/src/lib/components/send/SendInputAmount.svelte [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+	import { slide } from 'svelte/transition';
+	import { i18n } from '$lib/stores/i18n.store';
+	import type { BigNumber } from '@ethersproject/bignumber';
+	import { invalidAmount } from '$lib/utils/input.utils';
+
+	export let amount: number | undefined = undefined;
+	export let tokenDecimals: number | undefined = undefined;
+	export let label: string = $i18n.core.text.amount;
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/send/SendInputAmount.svelte:12]: the label is not hardcoded in the component using `$i18n.core.text.amount`

File: src/frontend/src/eth/components/send/SendAmount.svelte [hunk 9]
```diff
@@ -1,91 +1,81 @@
...
-	$: insufficientFunds = nonNullish(insufficientFundsError);
+	$: insufficientFunds = nonNullish(amountError);
 
 	const { feeStore: storeFeeData } = getContext<FeeContext>(FEE_CONTEXT_KEY);
 	const { sendTokenDecimals, sendBalance, sendTokenId } = getContext<SendContext>(SEND_CONTEXT_KEY);
 
-	const validate = () => {
+	const validate = (value: BigNumber) => {
 		if (invalidAmount(amount)) {
-			insufficientFundsError = undefined;
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/eth/components/send/SendAmount.svelte:30]: Reverted

File: src/frontend/src/lib/types/send.ts [hunk 10]
```diff
@@ -8,3 +8,7 @@ export interface TransferParams {
 	maxFeePerGas: bigint;
 	data?: string;
 }
+
+export class AmountAssertionError extends Error {}
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/types/send.ts:12]: Removed

File: src/frontend/src/eth/components/send/SendAmount.svelte [hunk 11]
```diff
@@ -1,91 +1,81 @@
...
+	let amountError: AmountAssertionError | undefined = undefined;
 
-	$: insufficientFunds = nonNullish(insufficientFundsError);
+	$: insufficientFunds = nonNullish(amountError);
 
 	const { feeStore: storeFeeData } = getContext<FeeContext>(FEE_CONTEXT_KEY);
 	const { sendTokenDecimals, sendBalance, sendTokenId } = getContext<SendContext>(SEND_CONTEXT_KEY);
 
-	const validate = () => {
+	const validate = (value: BigNumber) => {
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/eth/components/send/SendAmount.svelte:27]: Reverted

File: src/frontend/src/eth/components/send/SendAmount.svelte [hunk 12]
```diff
@@ -1,91 +1,81 @@
...
-
-			insufficientFundsError = undefined;
-			return;
 		}
 
 		// If ERC20, the balance of the token - e.g. 20 DAI - should cover the amount entered by the user
-		if (userAmount.gt($sendBalance ?? BigNumber.from(0n))) {
-			insufficientFundsError = $i18n.send.assertion.insufficient_funds_for_amount;
-			return;
+		else if (value.gt($sendBalance ?? BigNumber.from(0n))) {
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/eth/components/send/SendAmount.svelte:47]: Reverted

File: src/frontend/src/icp/components/send/IcSendAmount.svelte [hunk 13]
```diff
@@ -50,62 +43,42 @@
 				tokenDecimals: $tokenDecimals,
 				i18n: $i18n
 			});
-
-			if (nonNullish(amountError)) {
-				return;
+		} else if (isNetworkIdEthereum(networkId)) {
```
[the remark this answers is not in the document]
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/components/send/IcSendAmount.svelte:46]: Reverted

File: src/frontend/src/icp/components/send/IcSendAmount.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/icp/components/send/IcSendAmount.svelte:75]: Reverted

File: src/frontend/src/lib/components/send/SendInputAmount.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/send/SendInputAmount.svelte:33]: Corrected
